### PR TITLE
Stop telemetry when RCP disconnects

### PR DESCRIPTION
**Pull request by ryandoherty**
Works over BT & USB. Added a sort-of real 'disconnected' event to the RCP api object to make this easier.

### CODE REVIEW

File: autosportlabs/racecapture/api/rcpapi.py [hunk 1]
```diff
@@ -72,6 +72,8 @@ def __init__(self, **kwargs):
         self._enable_autodetect = Event()
         self._enable_autodetect.set()
 
+        self._disconnect_callback = kwargs.get('on_disconnect') if 'on_disconnect' in kwargs else None
```
Comment (brentpicasso) [inline autosportlabs/racecapture/api/rcpapi.py]: I think you'll get the desired effect if you just do self._disconnect_callback = kwargs.get('on_disconnect') - 

if the parameter isn't present the .get() returns None

File: main.py [hunk 2]
```diff
@@ -338,7 +338,7 @@ def build_preferences_view(self):
         preferences_view = PreferencesView(name='preferences', settings=self.settings, base_dir=self.base_dir)
         preferences_view.settings_view.bind(on_config_change=self._on_preferences_change)
         return preferences_view
-    
+    2
```
Comment (brentpicasso) [inline main.py]: syntax error?

File: main.py [hunk 3]
```diff
@@ -426,7 +426,11 @@ def rc_detect_win(self, version):
         if version.is_compatible_version():
             self.showStatus("{} v{}.{}.{}".format(version.friendlyName, version.major, version.minor, version.bugfix), False)
             self.dataBusPump.startDataPump(self._databus, self._rc_api)
-            self._status_pump.start(self._rc_api)        
+            self._status_pump.start(self._rc_api)
+
+            if self.settings.userPrefs.get_pref('preferences', 'send_telemetry') == "1" and self._telemetry_connection:
```
Comment (brentpicasso) [inline main.py]: just refreshed myself on bool() parsing in python. Would be nice to have a helper function in userPrefs to do something like:

userPrefs.is_pref_enabled(....  and would return a bool()

But not attached to it for this PR.

File: autosportlabs/racecapture/api/rcpapi.py (hunk 1 above)
Comment (ryandoherty) [inline autosportlabs/racecapture/api/rcpapi.py]: :+1:

File: main.py (hunk 2 above)
Comment (ryandoherty) [inline main.py]: Weird, no idea how that got there.

File: main.py (hunk 3 above)
Comment (ryandoherty) [inline main.py]: :+1:

**Comment by brentpicasso** [comment]
:sake:

**Pull request merged**